### ISSUE TITLE
CIF visualised in a malformed way due to the placement of the `_atom_sites_solution_primary` data item

### ISSUE THREAD

**Issue by vaitkus** [BobHanson/Jmol-SwingJS#32]
The issue manifests at least since version 14.32.73, but no earlier than version 14.29.17.
The issue is still present in version 14.32.77.

Steps to reproduce the issue:
1. Load the attached CIF file ([test-3500015.txt](https://github.com/BobHanson/Jmol-SwingJS/files/9870021/test-3500015.txt)) to Jmol.

Upon loading the file, the molecule is visualised in a malformed way with all of the atoms crammed together (see the image below).
![3500015-malformed](https://user-images.githubusercontent.com/5329163/198041177-6bcdfd18-14fe-4d3e-8f90-be22a79c88f8.png)

However, if the line with the `_atom_sites_solution_primary` data item is removed, commented out or placed as the first data item, the molecule is described in a proper way (see the image below). 
![3500015-not-malformed](https://user-images.githubusercontent.com/5329163/198047409-5e1db083-e7da-4b81-924f-d52b41ace5f5.png)

Most likely this is a CIF parsing issue since data items in a CIF file are allowed to appear in any order.

**Comment by nbehrnd** [comment]
The observation may be replicated with the current version of Jmol (14.32.78  2022-10-22 10:33) when loading the structure from Jmol's console, i.e.

```
load =cod/3500015;
```

The subsequent export from Jmol (`write 3500015.pdb;`) yields a model which, when loaded, equally shows this state of «too tightly packed van der Waals spheres».  Perhaps noteworthy, similar of .cif in proximity to the entry in question by deposition number

```load
load =cod/3500010;
load =cod/3500008;
```

yield a reasonable representation.  So there is something in the data of this special model Jmol has difficulties with.

**Comment by nbehrnd** [comment]
By replacement method, I observe Jmol struggles in case of *simultaneous* presence of the two lines

```
_atom_sites_solution_primary     direct
_atom_sites_solution_secondary   difmap
```

in the model datum.  The presence of either line `_atom_sites_solution_primary`, or `_atom_sites_solution_secondary` in level of `direct`, or `difmap` however does not cause a problem for Jmol.

**Comment by BobHanson** [comment]
Thanks -- this was a bug in CifReader skipping a line after processing
non-loop "atom_..." keys. Fixed for Jmol 14.32.79

**Comment by BobHanson** [comment]
Fixed

Jmol.___JmolVersion="15.2.79" // also 14.32.79

bug fix: ModelKit fixes for set picking rotateBond and hover labels when
ModelKit is not ON

*bug fix: CifReader can skip a line if loop processor is processing
non-loop keys starting with "atom_"*bug fix: write("OBJ") does not fail
gracefully
bug fix: Frank-click popup menu fails to open




On Wed, Oct 26, 2022 at 1:21 PM Robert Hanson ***@***.***> wrote:

> Thanks -- this was a bug in CifReader skipping a line after processing
> non-loop "atom_..." keys. Fixed for Jmol 14.32.79
>
>

-- 
Robert M. Hanson
Professor of Chemistry
St. Olaf College
Northfield, MN
http://www.stolaf.edu/people/hansonr


If nature does not answer first what we want,
it is better to take what answer we get.

-- Josiah Willard Gibbs, Lecture XXX, Monday, February 5, 1900

*We stand on the homelands of the Wahpekute Band of the Dakota Nation. We
honor with gratitude the people who have stewarded the land throughout the
generations and their ongoing contributions to this region. We acknowledge
the ongoing injustices that we have committed against the Dakota Nation,
and we wish to interrupt this legacy, beginning with acts of healing and
honest storytelling about this place.*
